### PR TITLE
Fix config.get for iterable sections

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Was using config.get method and realised it should take an iterable of str to handle nested config sections but the current implementation only works for a single section str.
Used the already existing get_sections method to handle this.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/core/config.py [hunk 1]
```diff
@@ -615,7 +615,11 @@ def get(
         self, val: str, section: Union[str, Iterable[str]] = "core", default: Any = None
     ):
         """Get a particular value from the config."""
-        return self._configs[section].get(val, default)
```
Comment (jpy-git) [inline src/sqlfluff/core/config.py:618]: would only work for if session a str

File: src/sqlfluff/core/config.py [hunk 2]
```diff
@@ -615,7 +615,11 @@ def get(
...
+        section_dict = self.get_section(section)
```
Comment (jpy-git) [inline src/sqlfluff/core/config.py:618]: handles nested config sections

File: src/sqlfluff/core/config.py [hunk 3]
```diff
@@ -615,7 +615,11 @@ def get(
...
+        if section_dict is None:
+            return default
```
Comment (jpy-git) [inline src/sqlfluff/core/config.py:620]: returns default if the section doesn't exist. Could alternatively raise a KeyError if preferred. Fine either way

**Comment by jpy-git** [comment]
@barrywhart could you take a look at this please, it's a quick bugfix for the config.get method

**Review by barrywhart — APPROVED** [review]

**Pull request merged**